collect/collect.go

```diff
@@ -1737,3 +1734,19 @@ func (i *InMemCollector) sendDecisions(decisionChan <-chan TraceDecision, interv
 		}
 	}
 }
+
+func (i *InMemCollector) recordQueueMetrics() {
```

Comment:
This should be part of some sort of waitgroup so we can wait for it to finish during shutdown. Otherwise you might see it fire after the Metrics subsystem has been stopped (assuming it does get stopped), which could create weird crashes on shutdown.